### PR TITLE
Consolidating references to storage apps table

### PR DESCRIPTION
**Pull request by maureensturgeon**
The changes here remove direct references to the storage apps table in pegasus and replace them with references to methods in `storage_apps.rb`. This is in preparation to move the table into dashboard DB and rename it `projects`. Minimizing the number of places that we reference the table name makes this transition easier.

(Swap out behind the flag started here: https://github.com/code-dot-org/code-dot-org/pull/45476)

## Links

- jira ticket: [LP-2266](https://codedotorg.atlassian.net/browse/LP-2266)

## Testing story
Tested that project pages, creating a new project and project-backed levels still work as expected.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -168,16 +168,20 @@ def index
   end
 
   def project_and_featured_project_fields
-    [
-      :storage_apps__id___id,
-      :storage_apps__storage_id___storage_id,
-      :storage_apps__value___value,
-      :storage_apps__project_type___project_type,
-      :storage_apps__published_at___published_at,
+    storage_apps_fields = prefix_storage_app_fields(%w(id___id storage_id___storage_id value___value project_type___project_type published_at___published_at))
```
Comment (jamescodeorg) [inline dashboard/app/controllers/projects_controller.rb:171]: Mostly curious, why was this change needed?

File: dashboard/app/models/user.rb [hunk 2]
```diff
@@ -2345,9 +2345,8 @@ def has_pilot_experiment?(pilot_name)
   private def soft_delete_channels
     return unless user_storage_id
 
-    channel_ids = PEGASUS_DB[:storage_apps].
-      where(storage_id: user_storage_id).
-      map(:id)
+    user_storage_apps = StorageApps.new(user_storage_id)
+    channel_ids = user_storage_apps.get_all_storage_ids
```
Comment (jamescodeorg) [inline dashboard/app/models/user.rb]: Should we rename `channel_ids` to reduce confusion here?

File: shared/middleware/helpers/storage_apps.rb [hunk 3]
```diff
@@ -347,10 +385,19 @@ def self.remix_ancestry(channel_id, depth: 1)
...
+    project_info = StorageApps.table.where(id: storage_app_id).first
     project_info[:abuse_score]
   end
 
+  # Returns storage apps with id in ids
+  def self.get_by_ids(ids)
+    StorageApps.table.where(id: ids)
+  end
+
+  def self.table
```
Comment (jamescodeorg) [inline shared/middleware/helpers/storage_apps.rb:396]: How much additional work is it to make this private?  (I'm guessing projects_list is going to biggest pain here, ok to defer until later but I think we'll want to make this private at some point.)

**Review by jamescodeorg — APPROVED** [review]

File: dashboard/app/controllers/projects_controller.rb (hunk 1 above)
Comment (maureensturgeon) [reply dashboard/app/controllers/projects_controller.rb:171]: I did this so that it'd be easier to swap out the table name prefix from `storage_apps` to `projects`

File: dashboard/app/models/user.rb (hunk 2 above)
Comment (maureensturgeon) [reply dashboard/app/models/user.rb]: Yes! Another perfect example of where the use of the term "channel" is confusing. I'll swap this out now, but I have another ticket https://codedotorg.atlassian.net/browse/LP-2272 where I intend to do a more focused sweep of the names we're using

File: shared/middleware/helpers/storage_apps.rb (hunk 3 above)
Comment (maureensturgeon) [reply shared/middleware/helpers/storage_apps.rb:396]: Ideally in the future we'll be using an ActiveRecord model for queries and we won't need to explicitly reference the table

File: dashboard/app/controllers/projects_controller.rb (hunk 1 above)
Comment (jamescodeorg) [reply dashboard/app/controllers/projects_controller.rb:171]: I see.  I find this a bit harder to read but hopefully this will all go away very soon so this looks good to me for now.  Ship it!
